Placeholder alignment options, clearButtonMode, option for showing/hiding of the password toggle and minor bug fixes.

sasojadrovski opened the pull request:
Added an option for setting the placeholder alignment.
Fixed an issue where the caret is improperly aligned the first time
`becomeFirstResponder` is called when the `textAlignment` property is
changed.

sasojadrovski committed:
Swift 3.1 updates.

Silenced warnings produced by Swift 3.1 syntax changes.
Added @discardableResult to NSLayoutConstraints functions.
Updated the ‘Example’ project file to the recommended settings.

sasojadrovski committed:
Placeholder alignment options.

Added an option for setting the placeholder alignment.

sasojadrovski committed:
Text alignment bug fix.

Fixes an issue where the caret is improperly aligned the first time
`becomeFirstResponder` is called when the `textAlignment` property is
changed.

sasojadrovski committed:
Removed unintentional warning.

Added @discardableResult to the overridden `becomeFirstResponder` to
prevent the warning.

jorgerebolloj commented:
@sasojadrovski Greetings, how can I configure the AnimatedTextInput for alignment to the center with the implementations that you implemented? Thanks in advance

sasojadrovski committed:
Exposed the `clearButtonMode` property.

Allowed changing the `clearButtonMode` property. Default is
`.whileEditing`.

sasojadrovski committed:
Improved the frames for textRect and editingRect.

Minor improvements to the `textRect` and `editingRect` frames. The text
was overlapping the `clearButton` or the `rightView`.

sasojadrovski committed:
Added an option for showing/hiding the password toggle.

Added an option for showing/hiding the password toggle. Unfortunately
this causes a minor API change since it is not possible to assign a
default value to a tuple.

victorBaro commented on AnimatedTextInput/Classes/AnimatedTextInput.swift:
```diff
@@ -157,8 +169,14 @@ open class AnimatedTextInput: UIControl {
-        let hintPosition = CGPoint(x: style.leftMargin, y: style.yHintPositionOffset)
-        let defaultPosition = CGPoint(x: style.leftMargin, y: style.topMargin + style.yPlaceholderPositionOffset)
+        let hintPosition = CGPoint(
+            x: placeholderAlignment != .natural ? 0 : style.leftMargin,
```
I'd rather see "positive" condition statements: 
`placeholderAlignment == .natural ? style.leftMargin : 0`

sasojadrovski replied:
I will take it into consideration in the future :) I agree with you!

victorBaro commented on AnimatedTextInput/Classes/AnimatedTextInput.swift:
```diff
@@ -157,8 +169,14 @@ open class AnimatedTextInput: UIControl {
+            y: style.yHintPositionOffset
+        )
+        let defaultPosition = CGPoint(
+            x: placeholderAlignment != .natural ? 0 : style.leftMargin,
```
Same

victorBaro commented on AnimatedTextInput/Classes/AnimatedTextInputFieldConfigurator.swift:
```diff
@@ -5,7 +5,7 @@ public struct AnimatedTextInputFieldConfigurator {
         case standard
         case email
-        case password
+        case password(toggleable: Bool)
```
I like that 😍

sasojadrovski replied:
Happy to hear that! 😄

victorBaro reviewed (approved):
Love the new additions :clap:
I only had a minor code style comment, but merging anyway. Pretty excited about all the new stuff. 
Many thanks for contributing.

sasojadrovski commented:
@victorBaro 
Thank you for the awesome library! I really like the way it is developed and structured. It is easily understandable and therefore easy to contribute to. I am more than happy that you liked and merged my pull request. It means a lot! 

Keep up the great work, and I will continue contributing in case I have more ideas about new features and/or improvements :)
